bpf/lib/trace.h

```diff
@@ -170,6 +170,11 @@ emit_trace_notify(enum trace_point obs_point, __u32 monitor)
 	return true;
 }
 
+struct trace_ctx {
+	enum trace_reason reason;
+	__u32 monitor;
```

Comment:
nit, I believe that this is what `monitor` means in the code today so we could potentially document that here now that there's a dedicated place for the definition.
```suggestion
	__u32 monitor; /* Monitor length for number of bytes to forward in trace message. 0 means do not monitor. */
```